packages/legend-query-builder/src/stores/workflows/ServiceQueryBuilderState.ts

```diff
@@ -77,7 +77,11 @@ export class ServiceQueryBuilderState extends QueryBuilderState {
     this.onServiceChange = onServiceChange;
     this.onExecutionContextChange = onExecutionContextChange;
 
-    if (service.execution instanceof PureSingleExecution) {
+    if (
+      service.execution instanceof PureSingleExecution &&
+      service.execution.mapping !== undefined &&
+      service.execution.runtime !== undefined
+    ) {
       this.mapping = service.execution.mapping.value;
```

Comment:
lets instead add an asserTrue within this case statement
assertTrue(service.execution.mapping && service.execution.runtime, 'Service queries without runtime/mapping are not supported')